sandbox.go

```diff
@@ -624,14 +644,9 @@ func (sb *sandbox) populateNetworkResources(ep *endpoint) error {
 		}
 	}
 
-	for _, gwep := range sb.getConnectedEndpoints() {
-		if len(gwep.Gateway()) > 0 {
-			if gwep != ep {
-				break
-			}
-			if err := sb.updateGateway(gwep); err != nil {
-				return err
-			}
+	if ep == sb.getGatewayEndpoint() {
```

Comment:
Please retain the check we had to not call updateGateway() if the current ep is not same as gwep. Not having the check will lead to unnecessary calls to updateGateway()

LGTM other than this one comment.